app/controllers/registrations_controller.rb

```diff
@@ -666,16 +735,18 @@ def refund_payment
       return redirect_to edit_registration_path(registration)
     end
 
-    # Backwards compatibility: We may at some point try to record a refund for a payment that was
-    #   - created before the introduction of receipts
-    #   - but refunded after the new receipts feature was introduced. Fall back to the old stripe_charge_id if that happens.
-    charge_id = payment.receipt&.stripe_id || payment.stripe_charge_id
+    if payment.receipt.nil?
+      # Backwards "compatibility": Before the introduction of receipts, in an old age when we were
+      #   hard-wired to a much older version of the Stripe SDK, we stored charge IDs directly in a legacy column.
+      #   This was so, so long ago that nobody should try and refund one of these payments anyways.
+      flash[:danger] = "The payment you're trying to refund is too old. Please refund from your Stripe dashboard directly."
+      return redirect_to edit_registration_path(registration)
+    end
 
-    refund_receipt = stripe_integration.issue_refund(charge_id, refund_amount)
+    refund_receipt = payment_account.issue_refund(payment.receipt, refund_amount)
```

Comment:
`refund_receipt` is a bit confusing - perhaps `refund_record`? (as `issue_refund` seems to be returning a PaymentRecord object)